build_tools/provision.py

```diff
@@ -0,0 +1,240 @@
+#!/usr/bin/env python
+
+import argparse
+import sys
+import os
+import shutil
+from fetch_artifacts import (
+    retrieve_base_artifacts,
+    retrieve_enabled_artifacts,
+    s3_bucket_exists,
+)
+from pathlib import Path
+import tarfile
+from tqdm import tqdm
+from _therock_utils.artifacts import ArtifactPopulator
+import requests
+from packaging.version import Version, InvalidVersion
+
+
+def log(*args, **kwargs):
+    print(*args, **kwargs)
+    sys.stdout.flush()
+
+
+def _untar_files(output_dir, destination):
+    """
+    Retrieves all tar files in the output_dir, then extracts all files to the output_dir
+    """
+    output_dir_path = Path(output_dir).resolve()
+    # In order to get better visibility on untar-ing status, tqdm adds a progress bar
+    log(f"Extracting {destination.name} to {output_dir}")
+    with tarfile.open(destination) as extracted_tar_file:
+        for member in tqdm(
+            iterable=extracted_tar_file.getmembers(),
+            total=len(extracted_tar_file.getmembers()),
+        ):
+            extracted_tar_file.extract(member=member, path=output_dir_path)
+    destination.unlink()
+
+
+def _create_output_directory(args):
+    """
+    If the output directory already exists, delete it and its contents.
+    Then, create the output directory.
+    """
+    output_dir_path = args.output_dir
+    log(f"Creating directory {output_dir_path}")
+    if os.path.isdir(output_dir_path):
+        log(
+            f"Directory {output_dir_path} already exists, removing existing directory and files"
+        )
+        shutil.rmtree(output_dir_path)
+    os.mkdir(output_dir_path)
+    log(f"Created directory {output_dir_path}")
+
+
+def _get_github_release_assets(release_tag, amdgpu_family, release_version):
+    """
+    Makes an API call to retrieve the release's assets, then retrieves the asset matching the amdgpu family
```

Comment:
Are these requests using authentication when available, e.g. via `GITHUB_TOKEN`? If not, they will be subject to the lower unauthenticated rate limit: https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28